tests/benchmark/bloatnet/test_bloatnet.py

```diff
@@ -0,0 +1,242 @@
+    """Calculate CREATE2 address from factory, salt, and init code hash."""
+    create2_input = b"\xff" + bytes(factory) + salt.to_bytes(32, "big") + init_code_hash
+    addr_bytes = keccak256(create2_input)[12:]
+    return Address(addr_bytes)
+
+
+@pytest.mark.valid_from("Prague")
+def test_bloatnet_balance_extcodesize(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    gas_benchmark_value: int,
+):
+    """
+    BloatNet test using BALANCE + EXTCODESIZE pattern.
+
+    This test:
+    1. Uses 100+ pre-deployed 24KB contracts
+    2. Calls BALANCE (cold) then EXTCODESIZE (warm) on each
+    3. Maximizes cache eviction by accessing many contracts
+    4. Runs with benchmark gas values (5M, 50M, 500M)
+    """
+    gas_costs = fork.gas_costs()
+
+    # Calculate gas costs
+    intrinsic_gas = fork.transaction_intrinsic_cost_calculator()(calldata=b"")
+
+    # Cost per contract access
+    cost_per_contract = (
+        3  # PUSH20 for address
+        + gas_costs.G_COLD_ACCOUNT_ACCESS  # Cold BALANCE (2600)
+        + gas_costs.G_BASE  # POP balance
+        + gas_costs.G_WARM_ACCOUNT_ACCESS  # Warm EXTCODESIZE (100)
+        + gas_costs.G_BASE  # POP code size
+    )
+
+    # Calculate how many contracts to access
+    available_gas = gas_benchmark_value - intrinsic_gas - 1000  # Reserve for cleanup
+    contracts_needed = int(available_gas // cost_per_contract)
+    num_contracts = min(contracts_needed, NUM_DEPLOYED_CONTRACTS)
+
+    # Log the calculation for transparency
+    if contracts_needed > NUM_DEPLOYED_CONTRACTS:
+        import warnings
+
+        warnings.warn(
+            f"Test needs {contracts_needed} contracts for "
+            f"{gas_benchmark_value / 1_000_000:.1f}M gas, "
+            f"but only {NUM_DEPLOYED_CONTRACTS} are deployed. "
+            f"Deploy {contracts_needed - NUM_DEPLOYED_CONTRACTS} more contracts "
+            f"for full test coverage.",
+            stacklevel=2,
+        )
+
+    # Generate attack contract with unrolled loop
+    attack_code = Bytecode()
+
+    # Pre-calculate all contract addresses using CREATE2
+    for i in range(num_contracts):
+        addr = calculate_create2_address(FACTORY_ADDRESS, i, INIT_CODE_HASH)
+        attack_code += (
+            Op.PUSH20[int.from_bytes(bytes(addr), "big")]
+            + Op.DUP1
+            + Op.BALANCE
+            + Op.POP
+            + Op.EXTCODESIZE
+            + Op.POP
+        )
+
+    # Success marker (benchmark tests typically run out of gas, but this is for smaller tests)
+    attack_code += Op.PUSH1[1] + Op.PUSH0 + Op.SSTORE
+
+    # Deploy attack contract
+    attack_address = pre.deploy_contract(code=attack_code)
+
+    # Attack transaction
+    tx = Transaction(
+        to=attack_address,
+        gas_limit=gas_benchmark_value,
+        sender=pre.fund_eoa(),
+    )
+
+    # Post-state: just verify attack contract exists
+    # Benchmark tests run out of gas, so no success flag
```

Comment:
For these kind of attacks OOG is fine, but if we do edit anything in state, it should not OOG because we have to ensure the writes are flushed to trie and not discarded (just a side note)

Reply:
In our cases for SLOAD, EXTCODE_X for now we never write to state. So I'm getting more and more convinced of removing them basically.

WDYT?

Reply:
Removing those read tests? They are definitely necessary. Although you do not write anything to state, it is not known beforehand that you do not write anything to state (besides: you actually *do* write something to state: namely, you deduct the gas*gasPrice from the tx sender and update the nonce, so you still need to know the gas spent on execution) so you have to execute these (disk read) operations anyways. We would still need to know the worst case behavior is also if there are only disk read operations. So I would not recommend to remove those.